### ISSUE TITLE
Add capability to upsample with .resample().ffill() using time series on either axis

### ISSUE THREAD

**Issue by mitches-got-glitches** [pandas-dev/pandas#40613]
It would be good to have the capability to upsample using `.resample()` and forward fill using `.ffill()` on either axis.

It seems to be blocked by the Assertion Error here, but I can' see a clear reason.
https://github.com/pandas-dev/pandas/blob/a20c169f8aee9c272953be43d89dfa82be5e186b/pandas/core/resample.py#L1153

```python
# Code to reproduce.
import pandas as pd
dt_idx = pd.period_range(start='2021-01-01', freq='M', periods=3).to_timestamp()
# Transpose to have time series on columns
df = pd.DataFrame(range(3), index=dt_idx).T
resampled = df.resample('W-Mon', axis=1).ffill()
```

There is a workaround below so this is issue is low priority.

```python
# Current workaround.
import pandas as pd
dt_idx = pd.period_range(start='2021-01-01', freq='M', periods=3).to_timestamp()
# Transpose to have time series on columns
df = pd.DataFrame(range(3), index=dt_idx).T
resampled = df.resample('W-Mon', axis=1).first().ffill(axis=1)
```

**Comment by jbrockmendel** [comment]
axis keyword in resample has been removed, closing.
